Homebrew 4.0.0 (#929)
Homebrew/brew.sh: _posts/2023-02-13-homebrew-4.0.0.md

---
title: 4.0.0
author: MikeMcQuaid
redirect_from: /blog/4.0.0/
---

Today, I'd like to announce Homebrew 4.0.0. The most significant change since 3.6.0 enables significantly faster Homebrew-maintained tap updates by migrating from Git-cloned taps to JSON downloads.

Major changes and deprecations since 3.6.0:

> - Using JSON files downloaded from [formulae.brew.sh](https://formulae.brew.sh) rather than local homebrew-core and homebrew-cask taps.

Reviewer: ```suggestion
- Relying on JSON files downloaded from [formulae.brew.sh](https://formulae.brew.sh) for package installation rather than local homebrew/core and homebrew/cask taps.
```

Author: @EricFromCanada mild edit as I like the rest but don't like "Relying on"

  - **Please note: this is the largest change we have made to our update process since we split Homebrew/brew and Homebrew/homebrew-core repositories. Please bear with us, there may be a few bumps**.
  - If you had previously set `HOMEBREW_NO_AUTO_UPDATE`, `HOMEBREW_NO_INSTALL_FROM_API` or `HOMEBREW_AUTO_UPDATE_SECS` to work around bugs or annoyances: please consider unsetting these and tweaking the values based on the new behaviour.
  - Unless you are developing formulae or casks, you can `brew untap homebrew/core` and `brew untap homebrew/cask` to save some space.

>   - [`brew update` will now be run automatically less often (every 24 hours rather than every 5 minutes)](https://github.com/Homebrew/brew/pull/14376) and these auto-updates will be much faster as they no longer need to perform the slow `git fetch` of the huge homebrew-core and homebrew-cask taps' Git repositories.

Reviewer: ```suggestion
  - [`brew update` will now be run automatically less often (every 24 hours rather than every 5 minutes)](https://github.com/Homebrew/brew/pull/14376) and these auto-updates will be much faster as they no longer need to perform the slow `git fetch` of the huge homebrew/core and homebrew/cask taps' Git repositories.
```

Author: Should probably add something like this to our prose style guide and/or vale checks and pick one way of doing so

  - [This is now the default behaviour so the `HOMEBREW_INSTALL_FROM_API` variable has been removed and is a no-op](https://github.com/Homebrew/brew/pull/14412).

>   - If you wish to opt-out of this behaviour change, you can [`export HOMEBREW_NO_INSTALL_FROM_API=1`](https://github.com/Homebrew/brew/pull/14375). Please investigate [`HOMEBREW_API_AUTO_UPDATE_SECS`](https://github.com/Homebrew/brew/pull/14589) first.

Reviewer 2: I suggest reversing the order of this sentence. You'll have a better chance of making the point you wanted to to people who stop reading. Will think of how to word it exactly.

  - This behaviour is automatically disabled when using [certain commands](https://github.com/Homebrew/brew/pull/14524) or in configurations that [mostly build from source](https://github.com/Homebrew/brew/pull/14480).
  - [`HOMEBREW_API_DOMAIN` can be set to use mirrors for](https://github.com/Homebrew/brew/pull/14494) [formulae.brew.sh](https://formulae.brew.sh).
- Work has begun to migrate from Google Analytics to a self-hosted InfluxDB instance. You can [help us test this beta feature by setting `export HOMEBREW_ANALYTICS_ENABLE_INFLUX=1`](https://github.com/Homebrew/brew/pull/13356).

> - [macOS `.pkg` files are generated by releases](https://github.com/Homebrew/brew/pull/14265). You can help us test this beta feature by [downloading the generated package artifact from the relevant GitHub Actions release events](https://github.com/Homebrew/brew/actions/workflows/build-pkg.yml?query=event%3Arelease).

Reviewer: ```suggestion
- [macOS `.pkg` files are generated for each Homebrew release](https://github.com/Homebrew/brew/pull/14265). You can help us test this beta feature by [downloading the generated package artifact from the relevant GitHub Actions release events](https://github.com/Homebrew/brew/actions/workflows/build-pkg.yml?query=event%3Arelease).
```

- [The `homebrew/ubuntu16.04:master` image has been deprecated](https://github.com/Homebrew/brew/pull/13819).
- [Various major release deprecations and disables](https://github.com/Homebrew/brew/pull/14382).

Other changes since 3.6.0 I'd like to highlight are the following:

- [`brew test` sets `PYTHONDONTWRITEBYTECODE` to improve performance](https://github.com/Homebrew/brew/pull/14338).
- [VSCode devcontainers are available in the Homebrew/brew repository for use with e.g. GitHub Codespaces](https://github.com/Homebrew/brew/pull/14027).

> - [`brew install --adopt` allows taking ownership of existing installed casks](https://github.com/Homebrew/brew/pull/14033).

Reviewer: ```suggestion
- [`brew install --adopt` allows taking ownership of existing installed cask artifacts](https://github.com/Homebrew/brew/pull/14033).
```

- [`brew install --dry-run` allows viewing what `brew install` will do before it is run](https://github.com/Homebrew/brew/pull/13706).
- [`brew docs` opens](https://github.com/Homebrew/brew/pull/13844) [docs.brew.sh](https://docs.brew.sh).
- [`--cask` is never required on Linux](https://github.com/Homebrew/brew/pull/14206).
- [`service do` blocks allow defining a `run` command per platform](https://github.com/Homebrew/brew/pull/14371).
- [`brew install` uses the local cache while installing dependencies from `pip`](https://github.com/Homebrew/brew/pull/14239).
- [`brew doctor` no longer complains about BitDefender](https://github.com/Homebrew/brew/pull/14300).
- [`brew install` will also suggest casks rather than just formulae when it fails to find the requested package](https://github.com/Homebrew/brew/pull/14240).
- [`brew readall` simulates all architecture and OS configurations for better reliability](https://github.com/Homebrew/brew/pull/13990).

> - [`git` partial clones with sparse checkouts are supported when downloading from `git`](https://github.com/Homebrew/brew/pull/14035).

Reviewer: ```suggestion
- [`git` partial clones with sparse checkouts are supported when downloading using `git`](https://github.com/Homebrew/brew/pull/14035).
```

- [We have discontinued WSL 1 support and recommend WSL 2 instead](https://github.com/Homebrew/brew/pull/14000).
- [`brew fetch` and `brew install` can automatically determine mirrors for `glibc-bootstrap` and PyPI resources](https://github.com/Homebrew/brew/pull/13875).

Finally:

- Homebrew is now on Mastodon: [@homebrew@fosstodon.org](https://fosstodon.org/@homebrew).
- [Our governance documents have been updated as part of our AGM](https://github.com/Homebrew/brew/pull/14482).
- [We have switched from HackerOne](https://github.com/Homebrew/brew/pull/14132) to [GitHub for vulnerability reporting](https://github.com/homebrew/brew/security/advisories/new).
- [Homebrew accepts donations through GitHub Sponsors](https://github.com/sponsors/Homebrew) and [still accepts donations through Patreon](https://www.patreon.com/homebrew). If you can afford it, please consider donating. If you'd rather not use GitHub Sponsors or Patreon (our preferred donation methods), [check out the other ways to donate in our README](https://github.com/homebrew/brew/#donations).

> Thanks to all our hard-working maintainers, contributors, sponsors and supporters for getting us this far.

Reviewer 3: Might this be a cool opportunity to post that picture from FOSDEM?

Author: @colindean you mean to further humiliate me in that stupid costume? I THINK NOT

(ok I'll add it)

Reviewer 3: 🤣
